### ISSUE TITLE
KubeCPUOvercommit is firing when using Jobs for batch processing

### ISSUE THREAD

**Issue by point9repeating** [kubernetes-monitoring/kubernetes-mixin#97]
We have a use case for a namespace in kubernetes where we submit a large number of Jobs to our cluster without expecting immediate scheduling. 

We've tested these workloads with a recently deployed kube-prometheus stack and have found out that the KubeCPUOvercommit fires whenever we submit a a large batch of jobs at once. I believe there are 2 issues here:
 * Pods stick around even when a Job completes successfully. kube-state-metrics will continue reporting metrics about that pod and this alert is assuming that those resource requests are still valid.
 * Even with auto-scaling enabled, we can reach peaks where we have more resource requests with pending jobs than available with whatever maximum number of nodes we've set. In our use case, we're ok with the scheduler working through all the Jobs as resources are freed up.

Here's the definition for the metric that KubeCPUOvercommit relies on.
```yaml
    - expr: |
        sum by (namespace, label_name) (
          sum(kube_pod_container_resource_requests_cpu_cores{job="kube-state-metrics"} and on(pod) kube_pod_status_scheduled{condition="true"}) by (namespace, pod)                                                                                                                  
        * on (namespace, pod) group_left(label_name)
          label_replace(kube_pod_labels{job="kube-state-metrics"}, "pod_name", "$1", "pod", "(.*)")
        )
      record: namespace_name:kube_pod_container_resource_requests_cpu_cores:sum
```

For the first issue, I believe ```kube_pod_container_resource_requests_cpu_cores``` will continue showing resource requests and ```kube_pod_status_scheduled{condition="true"}``` will have a value of ```1``` for any pod associated with a finished job. Perhaps we can join on another metric like ```kube_pod_status_phase{phase=~"Running|Pending"}```

**Comment by pvanderlinden** [comment]
Another scenario:
I submit large jobs (a job which spins up a large on demand dask cluster), which will mean that indeed we can't tolerate a node failure. Not sure how to handle this though, for know I have increased the delay before triggering to a few hours

**Comment by paulfantom** [comment]
I am closing the issue as it should be fixed with #646. If you still see the problem, please reopen.
